pkg/controllers/node/termination/controller.go

```diff
@@ -92,20 +94,39 @@ func (c *Controller) finalize(ctx context.Context, node *corev1.Node) (reconcile
 		return reconcile.Result{}, nil
 	}
 
-	nodeClaims, err := nodeutils.GetNodeClaims(ctx, c.kubeClient, node)
+	nodeClaim, err := nodeutils.NodeClaimForNode(ctx, c.kubeClient, node)
 	if err != nil {
-		return reconcile.Result{}, fmt.Errorf("listing nodeclaims, %w", err)
+		if nodeutils.IsDuplicateNodeClaimError(err) || nodeutils.IsNodeClaimNotFoundError(err) {
+			log.FromContext(ctx).Error(err, "failed to terminate node")
+			return reconcile.Result{}, nil
```

Comment:
Nit. 

Why don't we want to do anything if we find more then one nodeclaims? Are we expecting the customer to act in those cases?

Reply:
Yes, this means that an invariant was broken (e.g. a customer created a nodeclaim manually with the same provider ID). Making the termination controller tolerant to this significantly complicates things since the NodeClaim is the source of truth for where in the termination lifecycle we are. If there are multiple NodeClaims we have multiple sources of truth. Handling that isn't worth the extra complexity.

Reply:
> Yes, this means that an invariant was broken (e.g. a customer created a nodeclaim manually with the same provider ID)
Is this really an invariant we have taken? We have thus far created our nodeclaims to only link to one node, but this is not something we have definitively codified, no? To me this seems like a breaking change, since we have supported terminating multiple nodeclaims and have one providerID. To be fair, I don't think anyone should do this in practice, but was mainly thinking this could break some customers.

Reply:
I've discussed this pretty extensively offline with @jonathan-innis and @njtran, this was an overly defensive design back from the machine migration. This has never worked properly (e.g. TGP just picks the first NodeClaim) and is an invariant enforced by the NodeClaim creation mechanism. If we are in this state we should fail loudly instead of silently.